tests/vm_console_proxy_test.go

```diff
@@ -0,0 +1,135 @@
+package tests
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+)
+
+var _ = Describe("VM Console Proxy Operand", func() {
+	var (
+		clusterRoleResource        testResource
+		clusterRoleBindingResource testResource
+		serviceAccountResource     testResource
+		serviceResource            testResource
+		deploymentResource         testResource
+		configMapResource          testResource
+	)
+
+	BeforeEach(func() {
+		expectedLabels := expectedLabelsFor("vm-console-proxy", "vm-console-proxy")
+		clusterRoleResource = testResource{
+			Name:           "vm-console-proxy",
+			Resource:       &rbac.ClusterRole{},
+			ExpectedLabels: expectedLabels,
+		}
+		clusterRoleBindingResource = testResource{
+			Name:           "vm-console-proxy",
+			Resource:       &rbac.ClusterRoleBinding{},
+			ExpectedLabels: expectedLabels,
+		}
+		serviceAccountResource = testResource{
+			Name:           "vm-console-proxy",
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &core.ServiceAccount{},
+			ExpectedLabels: expectedLabels,
+		}
+		serviceResource = testResource{
+			Name:           "vm-console-proxy",
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &core.Service{},
+			ExpectedLabels: expectedLabels,
+		}
+		deploymentResource = testResource{
+			Name:           "vm-console-proxy",
+			Namespace:      strategy.GetVmConsoleProxyNamespace(),
+			Resource:       &apps.Deployment{},
+			ExpectedLabels: expectedLabels,
+		}
```

Comment:
For the update tests, these `testResource` structures need to have the `UpdateFunc` and `EqualsFunc` defined. They are used to do the update and then test that the resource was reverted.

Reply:
Done, thanks!